src/move.c

```diff
@@ -10,29 +75,86 @@ static float cell_shift_micrometers = 42000;
  */
 void move_straight_out_of_cell(void)
 {
-	int32_t starting_position = get_encoder_average_micrometers();
+	int32_t target_distance;
 
+	target_distance = get_encoder_average_micrometers() +
+			  (int32_t)(CELL_DIMENSION * MICROMETERS_PER_METER) -
+			  cell_shift_micrometers;
 	set_target_angular_speed(0.);
-	set_target_linear_speed(.5);
-	while (get_encoder_average_micrometers() - starting_position <
-	       CELL_DIMENSION * MICROMETERS_PER_METER - cell_shift_micrometers)
+	set_target_linear_speed(linear_speed);
+	while (get_encoder_average_micrometers() < target_distance)
 		;
-	led_left_toggle();
+	entered_next_cell();
 }
 
 /**
- * @brief Move straight into the next cell.
+ * @brief Move straight and stop at the end of the current cell.
  */
-void move_straight(void)
+void move_straight_stop_end(void)
 {
-	int32_t starting_position = get_encoder_average_micrometers();
+	int32_t target_distance;
+	uint32_t target_ticks;
 
+	target_distance = current_cell_start_micrometers +
+			  (int32_t)(CELL_DIMENSION * MICROMETERS_PER_METER) -
+			  required_micrometers_to_stop();
 	set_target_angular_speed(0.);
-	set_target_linear_speed(.5);
-	while (get_encoder_average_micrometers() - starting_position <
-	       CELL_DIMENSION * MICROMETERS_PER_METER)
+	set_target_linear_speed(linear_speed);
+	target_ticks = required_ticks_to_stop();
+	while (get_encoder_average_micrometers() < target_distance)
 		;
-	led_left_toggle();
+	set_target_linear_speed(.0);
+	target_ticks += get_clock_ticks();
+	while (get_clock_ticks() < target_ticks)
+		;
+	entered_next_cell();
+}
+
+/**
+ * @brief Move straight and stop when the head would touch the front wall.
+ */
+void move_straight_stop_head_front_wall(void)
```

Comment:
Maybe in the future would be clear one function with different inputs depending on move_straight type as they are sharing the same calculations.

Reply:
Yeah, `move.c` will require refactorization, but it is currently under heavy development, so not worth it I think. :wink: